### DISCUSSION TITLE
《宝可梦》第四世代汉化修正

### DISCUSSION THREAD

**Discussion by giscus-xzonn** [Xzonn/PokemonChineseTranslationRevise#18] [General]
# 《宝可梦》第四世代汉化修正

https://blog.www.localhost/PokemonChineseTranslationRevise/

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Apollonian 评论于 2019-04-14 13:00:40 (UTC+8)*****

首先很感谢重译，良心之作。其次好像所有的"幸福"都被替换成"幸福蛋"了，不知道是不是笔误。

*****Xzonn 回复于 2019-04-27 16:04:57 (UTC+8)*****

感谢反馈，最新版本已修正。这是我在批量替换的时候没有注意导致的。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Apollonian 评论于 2019-06-06 15:33:33 (UTC+8)*****

我试了下0.9.1，不知道为什么都成乱码了。上次0.9.0没有这个问题，我操作和打0.9.0的补丁也没啥区别……

![https://d47jbcq60tnr6.cloudfront.net/201966/32531-12td56g.s4hn.PNG](https://d47jbcq60tnr6.cloudfront.net/201966/32531-12td56g.s4hn.PNG)

*****Xzonn 回复于 2019-07-20 17:48:05 (UTC+8)*****

感谢反馈，我好久没看评论了……查了一下是当时我把码表和《心金》的码表弄混了，现在已经修正了。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****macha 评论于 2019-07-07 19:31:02 (UTC+8)*****

好奇你是用什么工具制作的？

*****Xzonn 回复于 2019-07-20 17:48:51 (UTC+8)*****

参考了一些前人写好的工具，然后自己写了代码修改的。

*****Apollonian 回复于 2019-07-26 01:17:41 (UTC+8)*****

啊我也好奇。我要是有工具……好吧还有技术和时间……我就是弄gen5了。

*****Xzonn 回复于 2019-07-26 22:38:50 (UTC+8)*****

等我有时间整理下代码之后开源吧……最初写的代码因为是自用所以比较乱。

*****Apollonian 回复于 2019-07-27 00:00:14 (UTC+8)*****

我试了下0.9.2，有些小字符是乱码：  
心齐湖的齐  
草苗龟的龟  
湖底深处的深  
沉默的默  
三地鼠的鼠  
小火龙的龙  
霹雳电球的雳  
长鼻叶的鼻  
黄  
懒人獭的獭  
毒蔷薇的蔷薇  
猫鼬斩的鼬  
天秤偶的秤  
气闸的闸  
玛纳菲的菲  
腹鼓的鼓  
龙之俯冲的俯  
暂时只发现这么多。

*****Xzonn 回复于 2019-07-27 23:54:22 (UTC+8)*****

感谢回馈……我回去翻翻代码……

*****Apollonian 回复于 2019-07-28 13:47:08 (UTC+8)*****

老实说，不知道有没有可能干脆把Gen4和Gen5的几个游戏统一成同一个码表。更准确的说，是统一成黑白2的码表，因为黑白2用的好像是UTF-8

*****Xzonn 回复于 2019-07-30 22:20:53 (UTC+8)*****

Gen4统一成一个码表有可能，但前提是需要获取旧汉化版使用的码表，这个我目前还是人力统计的……Gen5我还没试过，但我觉得不太可能是UTF-8编码，倒是有可能是Unicode。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****MIKUsama 评论于 2019-08-14 08:56:47 (UTC+8)*****

補丁載點掛了

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****k90单反 评论于 2020-03-24 22:30:18 (UTC+8)*****

大佬你好，我尝试了一下你改后的ROM，真的很赞 :smile: 不过有几个问题想提一下  
开机画面里的logo还是一开始的版本，和最后新的那个不一致，见图12（当然我猜测是因为直接用了贴吧的图，没有psd格式可以抽取图层导致的。同样关于这个logo吹毛求疵一下，感觉用贴吧的图看上去边框有些锯齿感，而用来对比的原版边框就光滑得多，这一点心金魂银尤为明显，特别是心金周围还能看到一些白色部分，白金虽然不明显也有些锯齿感。所以可能的话不知道能否联系上logo的制作者要来psd格式，这样应该能处理得完美些）  
然后是“？”这个标点似乎间距太大，导致图鉴里面如果遇到但没抓过的宝可梦，身高体重那里显示不完整，可见图3  
如果大佬愿意修复一下就好了 :smile: 感谢大佬为宝可梦系列汉化的贡献

![https://d47jbcq60tnr6.cloudfront.net/2020324/14824-1hnaqkg.r7qei.jpg](https://d47jbcq60tnr6.cloudfront.net/2020324/14824-1hnaqkg.r7qei.jpg)

![https://d47jbcq60tnr6.cloudfront.net/2020324/17001-1efovte.m9bqj.jpg](https://d47jbcq60tnr6.cloudfront.net/2020324/17001-1efovte.m9bqj.jpg)

![https://d47jbcq60tnr6.cloudfront.net/2020324/14561-19755ty.2htv.jpg](https://d47jbcq60tnr6.cloudfront.net/2020324/14561-19755ty.2htv.jpg)

*****Xzonn 回复于 2020-03-25 15:40:44 (UTC+8)*****

感谢反馈，前两个图是漏了，最后那个图下个版本尝试把问号改成英文问号。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****裂石响惊弦 回复于 2020-04-01 11:03:15 (UTC+8)*****

其他页面看到了

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****MrVilla 评论于 2020-04-05 09:51:08 (UTC+8)*****

非常感谢对汉化的付出，没想到过去10年了还能玩到官方译名汉化版，感动……  
不过在玩心金魂银的时候也发现了几处比较不顺的翻译，可能是当时翻译比较仓促。  
第一个问题：对树使用的应该是头锤，不过显示使用的是头槙。  
第二个问题：在宝可梦培育屋，培育老奶奶问的是“想培育哪个呢？”，感觉此处翻译成“想要培育宝可梦么？”会更好。因为选择了“是”选项以后，下一句对话是“要培育哪个呢”。  
第三个问题：在桔梗市，使用特定短语，短语哥哥会送你三个蛋，分别是咩利羊、乌波、熔岩虫的蛋，这里的翻译顺序有些错乱。  
以上三处问题真的希望有时间能修复~太感谢你的付出了。

![https://d47jbcq60tnr6.cloudfront.net/202045/17720-4b4pnd.r0if5.jpg](https://d47jbcq60tnr6.cloudfront.net/202045/17720-4b4pnd.r0if5.jpg)

![https://d47jbcq60tnr6.cloudfront.net/202045/19473-2f1uh7.kux0u.jpg](https://d47jbcq60tnr6.cloudfront.net/202045/19473-2f1uh7.kux0u.jpg)

![https://d47jbcq60tnr6.cloudfront.net/202045/17720-nmiccc.j8w6f.jpg](https://d47jbcq60tnr6.cloudfront.net/202045/17720-nmiccc.j8w6f.jpg)

*****Xzonn 回复于 2020-04-05 15:53:41 (UTC+8)*****

感谢反馈，第一个是字库问题，后两个是原有翻译寄存在的问题，下个版本修复。

*****MrVilla 回复于 2020-04-05 17:08:28 (UTC+8)*****

感谢，期待修正版~  
顺便一提，我发现了门口科学胖子的彩蛋了哈哈。

*****Xzonn 回复于 2020-04-06 18:10:40 (UTC+8)*****

那个胖子实际上是用来标识版本号的（

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****yggdra 评论于 2020-04-06 15:39:15 (UTC+8)*****

围观了一下，发现HGSS-718md，板木老人→板木老大；“怪物盒”这个词有毛病，ぼくの　ポケモンいれ　からっぽ 我的宝可梦都用光了？

*****Xzonn 回复于 2020-04-06 18:09:53 (UTC+8)*****

感谢反馈，查了一下都是历史遗留问题（甩锅），坂木老大没啥问题，“ぼくの　ポケモンいれ　からっぽ”这句我没太看懂，和前后文联系起来猜测是说钓鱼的盒子空了？我觉得可以翻译成“我（钓到）的宝可梦都跑了！”

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****YUHJIS 评论于 2020-04-13 22:23:21 (UTC+8)*****

这个情况怎么解决。

![https://d47jbcq60tnr6.cloudfront.net/2020413/30788-1m232t2.q3s2i.jpg](https://d47jbcq60tnr6.cloudfront.net/2020413/30788-1m232t2.q3s2i.jpg)

*****藩篱 回复于 2020-04-25 10:38:02 (UTC+8)*****

你是从已经打过补丁的版本再次打新补丁了吗

*****？？？ 回复于 2020-04-29 15:51:55 (UTC+8)*****

我打其他补丁都正常，打心金的补丁也会这样

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****？？？ 评论于 2020-04-29 15:52:34 (UTC+8)*****

其他补丁都正常，但是打心金补丁会报错

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****？？？ 评论于 2020-04-29 17:55:09 (UTC+8)*****

再反馈一下，白金里对战开拓区电脑旁边有个可以查看宝可梦个体值的npc，和他对话他只会显示半句话，“请把你认为”后面的话不显示

![https://d47jbcq60tnr6.cloudfront.net/2020429/16700-12ziv99.s6a1.jpg](https://d47jbcq60tnr6.cloudfront.net/2020429/16700-12ziv99.s6a1.jpg)

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****？？？ 评论于 2020-04-29 18:32:14 (UTC+8)*****

再反馈一个，魂银进游戏之后对话框黑色，直到选性画面别就卡住了

![https://d47jbcq60tnr6.cloudfront.net/2020429/25193-meuqgs.yyalp.jpg](https://d47jbcq60tnr6.cloudfront.net/2020429/25193-meuqgs.yyalp.jpg)

![https://d47jbcq60tnr6.cloudfront.net/2020429/25193-1p7ab0u.ns0l.jpg](https://d47jbcq60tnr6.cloudfront.net/2020429/25193-1p7ab0u.ns0l.jpg)

*****？？？ 回复于 2020-04-29 18:32:41 (UTC+8)*****

直到选性别画面就卡住了

*****Spade 回复于 2020-04-29 19:50:57 (UTC+8)*****

不用看了你的rom不是干净的，我完全没问题

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****好人 评论于 2020-05-08 14:19:10 (UTC+8)*****

白金1.3.0版开头动画显示的还是口袋妖怪

![https://d47jbcq60tnr6.cloudfront.net/202058/1764-1qg06kh.3pn4.jpg](https://d47jbcq60tnr6.cloudfront.net/202058/1764-1qg06kh.3pn4.jpg)

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****前尘 评论于 2020-05-27 22:27:46 (UTC+8)*****

发现两个小问题，给碎岩术的胖子字句是碎石术，第一个记录电话的短裤小子说（我会打给你），出现了（我回答给你），会打和回答，估计是输入法的问题

*****前尘 回复于 2020-05-27 22:35:13 (UTC+8)*****

开始刷这款游戏，新游戏玩不下，每年都要拿出来玩金银，正式的汉化玩起来感觉很好，打算入个ndsi来玩

*****Xzonn 回复于 2021-06-28 13:34:19 (UTC+8)*****

感谢反馈，已修复。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****NO NICKNAME 评论于 2020-09-23 12:53:04 (UTC+8)*****

珍珠 0.9b 标题是口袋妖怪  劲敌的名字备选里没有阿驯

![https://d47jbcq60tnr6.cloudfront.net/2020923/18925-4j2m75.09aad.jpg](https://d47jbcq60tnr6.cloudfront.net/2020923/18925-4j2m75.09aad.jpg)

![https://d47jbcq60tnr6.cloudfront.net/2020923/15838-bw2f44.54l5.jpg](https://d47jbcq60tnr6.cloudfront.net/2020923/15838-bw2f44.54l5.jpg)

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****我會在繁星這頭等你 评论于 2021-02-20 21:24:09 (UTC+8)*****

所有的补丁都是这个问题，请问怎么回事？英语又渣，实在不知道怎么解决。。。

![https://d47jbcq60tnr6.cloudfront.net/2021220/15295-cxx4yv.ir93u.png](https://d47jbcq60tnr6.cloudfront.net/2021220/15295-cxx4yv.ir93u.png)

![https://d47jbcq60tnr6.cloudfront.net/2021220/3727-sqgwct.oox4.png](https://d47jbcq60tnr6.cloudfront.net/2021220/3727-sqgwct.oox4.png)

*****O口O！ 回复于 2021-06-13 01:32:24 (UTC+8)*****

我也是这样，你解决了吗？

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Kiseki 评论于 2021-03-17 22:51:50 (UTC+8)*****

白金1.4.0发现几个文本错误  
p1.帷幕百货b1与这个npc对话，感觉应该是地下商场。  
p2、p3.帷幕百货3楼这两个售货员npc的取消选项。

![https://d47jbcq60tnr6.cloudfront.net/2021317/3030-5xi91h.tow5n.jpg](https://d47jbcq60tnr6.cloudfront.net/2021317/3030-5xi91h.tow5n.jpg)

![https://d47jbcq60tnr6.cloudfront.net/2021317/15295-z2i746.3eyrr.jpg](https://d47jbcq60tnr6.cloudfront.net/2021317/15295-z2i746.3eyrr.jpg)

![https://d47jbcq60tnr6.cloudfront.net/2021317/6263-7z9su2.cmeo8.jpg](https://d47jbcq60tnr6.cloudfront.net/2021317/6263-7z9su2.cmeo8.jpg)

*****Xzonn 回复于 2021-06-28 13:20:51 (UTC+8)*****

感谢反馈，已修复。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****归零者 评论于 2021-03-26 00:56:51 (UTC+8)*****

请问心金1.3补丁打完赤红or二周目联盟后均无法正确读取存档？（只能读取上一次存档）

![https://d47jbcq60tnr6.cloudfront.net/2021325/16529-dvzeip.w2zyp.png](https://d47jbcq60tnr6.cloudfront.net/2021325/16529-dvzeip.w2zyp.png)

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****非节能型照明设备 评论于 2021-04-26 07:40:20 (UTC+8)*****

麻烦修图的巨巨修下“心金/魂银”标题边上的白色毛刺儿啊！如果不方便，让我来修也可以…

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****非节能型照明设备 评论于 2021-04-26 21:20:29 (UTC+8)*****

还有这个洛奇亚的尾巴少了几个像素…

![https://d47jbcq60tnr6.cloudfront.net/2021426/29255-jubwfa.6yc4c.png](https://d47jbcq60tnr6.cloudfront.net/2021426/29255-jubwfa.6yc4c.png)

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****尚弦月 评论于 2021-06-09 15:40:08 (UTC+8)*****

心金的补丁打不上，报错An error has occurred: xdelta3: target window checksum mismatch: XD3_INVALID_INPUT  
是我的版本不对吗？我下的是acg汉化组的，有没有好心人给个能打补丁的或者打好的心金

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****憨憨 评论于 2021-06-13 20:58:03 (UTC+8)*****

大佬您好，您的网站在微软edge浏览器上无法访问，我现在用的是IE浏览器，请您修复一下吧！

*****Xzonn 回复于 2021-06-28 13:12:09 (UTC+8)*****

请用最新版本的 Edge。我不考虑 IE 兼容性。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Xzonn 评论于 2021-06-28 13:18:49 (UTC+8)*****

此前打补丁如有报错问题，请尝试使用最新版本的补丁及工具。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****zero10000 评论于 2021-07-10 14:24:45 (UTC+8)*****

希望可以像上个版本那样放出直接升级补丁，不用再用原版rom做基础rom

*****Xzonn 回复于 2021-07-12 09:54:42 (UTC+8)*****

您好，1.4.1 版本的补丁可以直接用于原版 ROM，也可以用于之前版本的已修正的 ROM。

*****時風鐮夜 回复于 2021-07-15 14:40:28 (UTC+8)*****

可是我用白金ROM试了一下用1.3跟1.4下去打1.4.1补丁，校验出来的CRC32都跟说明文档上的不一样

*****時風鐮夜 回复于 2021-07-15 14:42:51 (UTC+8)*****

1.3跟1.4的ROM CRC32已经确认过跟之前旧版的说明文档上的CRC32是一致的

*****Xzonn 回复于 2021-07-18 12:16:22 (UTC+8)*****

我给出来的 CRC32 是拿 ACG 汉化版打补丁得到的 CRC32，因为之前的工具一直有人反馈只要输入的 ROM 不一样就没法打补丁，所以我换用了新的工具，原理和之前不同，因此暂时只能保证用 ACG 汉化版 ROM 得到的 ROM 的 CRC32 与文档中给出的一致，无法保证其他 ROM 得到的 ROM 的 CRC32与文档一致；但 ROM 的有效内容应该是完全一样的。另外目前还是过渡版本，我会再尝试一下能不能尽可能让各种 ROM 得到的结果保持一致。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****EiMos 评论于 2021-07-13 02:09:35 (UTC+8)*****

卧槽！没想到收集宝可梦历代资源的过程中居然能找到这个神仙网站来，太强了！先收藏了  
话说当年我们当年群星SP也想做这个来着，安老板提出要用ACG版改，但没人动手，ACG汉化版的HGSS文本至今还躺在群共享里233333

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Vasopressin 评论于 2021-07-31 22:29:10 (UTC+8)*****

部分特性效果在后续世代有改动，用后续世代的官方翻译会导致特性效果说明与实际不符。例如：结实、避雷针、引水。  
还有很多特性的说明原文没有译文那么详细，大部分都是省略了具体数值，GF当年这样写应该是有用意的（保持神秘感，不希望太详细的数字影响游戏乐趣），这些说明是不是也是按照原文翻译比较好？例如：愤怒穴位、贪吃鬼、耐热、预知梦、单纯等等。

*****Xzonn 回复于 2021-08-19 21:55:03 (UTC+8)*****

谢谢提醒，因为这部分我是直接导入的最新的特性说明文本，忘了校对，之后我会对着原文再看一遍。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Green 评论于 2021-09-09 16:10:47 (UTC+8)*****

1.4.1 好像用不了之前的手机金手指dat（usrcheat.dat）  
1.4 就可以

*****Green 回复于 2021-09-09 16:29:23 (UTC+8)*****

就是20210316那个dat，电脑上DeSmuSE模拟器都可以用，不管是1.4还是1.41。  
然后手机上Drastic模拟器，1.4可以搜出来，1.41就是空白。

*****Xzonn 回复于 2021-09-13 18:41:02 (UTC+8)*****

因为 1.4.1 版本是最新发布的，如果需要金手指可以手动修改，可参考：https://xzonn.top/posts/TWiLightMenu-Guide.html

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****刘能胜 评论于 2021-10-25 19:45:22 (UTC+8)*****

哇好厉害的大佬！！！

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Ming Chen 评论于 2021-11-23 04:47:16 (UTC+8)*****

請教一下，我已經用了補丁修改了原本的ROM，但存檔還是放不進您提供的PKHex，上面寫我嘗試加載不支持的文件類型，我是有哪個步驟漏掉了嗎?

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****p***a 评论于 2021-12-25 14:04:54 (UTC+8)*****

新修版的字体尺寸增大了不少，但显示效果感觉不怎么好，一个大一个小，左边粗右边细的，还没有原来的美观。

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****mikumikumi?!! 评论于 2022-04-08 11:45:39 (UTC+8)*****

太感谢大佬了！！！谢谢大佬的付出！微薄的言语表明不了我对大佬的感激之情！宝可梦有您如虎添翼@xzonn

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Scott 评论于 2022-06-06 18:27:00 (UTC+8)*****

请问下 我用了打补丁的工具 可是输出的nds文件是0kb的空文件 这个怎么解决啊

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****‭ 评论于 2022-07-05 17:21:13 (UTC+8)*****

我只想知道一件事，用以前的493补丁打在这个汉化版上精灵名字会不会乱码？

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Konyaka 评论于 2023-01-19 00:40:25 (UTC+8)*****

问题反馈，心金打上补丁后，图鉴信息都变成魂银版本的了

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Q.mi 评论于 2023-07-07 17:14:41 (UTC+8)*****

有没有只汉化内容，不用汉化捕抓的宝可梦

**Comment by Kumiri** [comment]
1.5版本能直接升级2.0吗？还是“2.0.0版本及以后版本的补丁仅适用于日文版ROM”？那日文ROM还得留备份，不然一直需要在网上寻找ROM文件……如果1.5能直接升2.0拿手里YYJoy汉化组ROM还有救……主要是不知道是不是在老男人游戏官网找到原版ROM就是能用的？

**Reply by Xzonn** [reply]
> 1.5版本能直接升级2.0吗？

不能。补丁工具会提示 ROM 的校验码是否正确。

**Reply by Kumiri** [reply]
好吧，还是老老实实备份ROM……还是要谢谢解惑。

**Comment by sadouxi96** [comment]
作者你好。
我使用的是打了2.0补丁的心金ROM（日版正确校验），但是打完以后我发现训练师卡片的右上角不显示星星（正常我应该有三颗星），而日版的训练师卡可以正常显示星星。
这是不是bug？

**Reply by Xzonn** [reply]
已在 2.1.0 版本修复。

**Reply by sadouxi96** [reply]
多谢！
